src/main/java/com/zendesk/maxwell/row/RowMapDeserializer.java

```diff
@@ -0,0 +1,120 @@
+package com.zendesk.maxwell.row;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ValueNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class RowMapDeserializer extends StdDeserializer<RowMap> {
+  private static ObjectMapper mapper;
+
+  public RowMapDeserializer() {
+    this(null);
+  }
+
+  public RowMapDeserializer(Class<?> vc) {
+    super(vc);
+  }
+
+  @Override
+  public RowMap deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+    JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+
+    JsonNode type = node.get("type");
+    JsonNode database = node.get("database");
+    JsonNode table = node.get("table");
+    JsonNode ts = node.get("ts");
+    JsonNode xid = node.get("xid");
+    JsonNode data = node.get("data");
+    JsonNode oldData = node.get("old");
+
+    RowMap rowMap = new RowMap(
+        type == null ? null : type.asText(),
+        database == null ? null : database.asText(),
+        table == null ? null : table.asText(),
+        ts == null ? null : ts.asLong(),
```

Comment:
these 4 fields should always be there in the JSON.  Rather than passing null and having unexpected things happen down the road, better to throw some kind of parse-exception if they don't exist.